### ISSUE TITLE
Write to database from a separate thread

### ISSUE THREAD

**Issue by andrewjw1995** [staff0rd/entityframework-logging#2]
It might not be such a good idea to call Context.SaveChanges() with every log message, as that blocks the current thread.

**Comment by staff0rd** [comment]
Yeah - if you're logging a whole lotta stuff to the db, it's probably better that a buffer or similar is implemented.
